src/control/common/file_utils.go

```diff
@@ -257,3 +258,73 @@ func FindBinary(binName string) (string, error) {
 
 	return adjPath, nil
 }
+
+// CpFile copies a file from src to dst.
+func CpFile(src, dst string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	out, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Copy the Directory from source to destination.
+func CpDir(source string, dest string) error {
+	// get properties of source dir
+	sourceinfo, err := os.Stat(source)
+	if err != nil {
+		return errors.Wrap(err, "unable to get FileInfo structure")
+	}
+
+	// create dest dir
+	err = os.MkdirAll(dest, sourceinfo.Mode())
+	if err != nil {
+		return errors.Wrap(err, "unable to create destination Folder")
+	}
+
+	directory, _ := os.Open(source)
+	objects, err := directory.Readdir(-1)
+
+	for _, obj := range objects {
+		sourceFile := source + "/" + obj.Name()
+		destinationFile := dest + "/" + obj.Name()
+
+		if obj.IsDir() {
+			// create sub-directories - recursively
+			err = CpDir(sourceFile, destinationFile)
+			if err != nil {
+				return errors.Wrap(err, "unable to Copy Dir")
+			}
+		} else {
+			// perform the file copy
+			err = CpFile(sourceFile, destinationFile)
+			if err != nil {
+				return errors.Wrap(err, "unable to Copy File")
+			}
+		}
+
+	}
+	return nil
+}
+
+// Check if file or directory that starts with . which is hidden
+func IsHidden(filename string) bool {
+	if filename[0:1] == "." {
```

Comment:
This will panic if `filename` is empty. Should probably add a `filename != "" &&` before this.